src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandler.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Arrays;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.FeePayment;
+
+@Component
+public class FeePaymentHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final FeePayment<AsylumCase> feePayment;
+    private final boolean isfeePaymentEnabled;
+
+    public FeePaymentHandler(
+            @Value("${featureFlag.isfeePaymentEnabled}") boolean isfeePaymentEnabled,
+            FeePayment<AsylumCase> feePayment
+    ) {
+        this.feePayment = feePayment;
+        this.isfeePaymentEnabled = isfeePaymentEnabled;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && isfeePaymentEnabled
+                &&
+                Arrays.asList(
+                   Event.SUBMIT_APPEAL
+                ).contains(callback.getEvent()
```

Comment:
Probably unnecessary to go through a list and call `contains()` on it.  Just want to check the `Event` in the `callback` == `SUBMIT_APPEAL`

Reply:
Yes - agreed and corrected.